mlkem/poly.h

```diff
@@ -59,7 +59,11 @@ typedef struct
 #pragma CPROVER check push
 #pragma CPROVER check disable "unsigned-overflow"
 #endif
-static INLINE uint32_t scalar_compress_d1(uint16_t u)
+/* TODO: This is marked STATIC_INLINE_TESTABLE to allow call-by-contract,
+   in CBMC, which does not seem to work at the moment. Once that's fixed,
+   this should be removed. */
+STATIC_INLINE_TESTABLE
+uint32_t scalar_compress_d1(uint16_t u)
```

Comment:
Can you elaborate why this is needed now? It seems to be working fine in CI?

Reply:
See updated version